src/flow_solver/flow_solver_cases/non_periodic_cube_flow.cpp

```diff
@@ -1,25 +1,70 @@
+                Physics::PhysicsFactory<dim,nstate,double>::create_Physics(parameters_input));
+}
 
 template <int dim, int nstate>
 std::shared_ptr<Triangulation> NonPeriodicCubeFlow<dim,nstate>::generate_grid() const
 {
-    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (this->mpi_communicator); // Mesh smoothing is set to none by default.
+    //std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (this->mpi_communicator); // Mesh smoothing is set to none by default.
+        std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (
+    #if PHILIP_DIM!=1
+                this->mpi_communicator
+    #endif
+        );
+
+    bool use_number_mesh_refinements = false;
+    if(this->all_param.flow_solver_param.number_of_mesh_refinements>0)
+        use_number_mesh_refinements = true;
     
-    const unsigned int number_of_refinements = this->all_param.flow_solver_param.number_of_mesh_refinements;
+    const unsigned int number_of_refinements = use_number_mesh_refinements ? this->all_param.flow_solver_param.number_of_mesh_refinements 
+                                                                           : log2(this->all_param.flow_solver_param.number_of_grid_elements_per_dimension);
+
     const double domain_left = this->all_param.flow_solver_param.grid_left_bound;
     const double domain_right = this->all_param.flow_solver_param.grid_right_bound;
     const bool colorize = true;
     
+    using flow_case_enum = Parameters::FlowSolverParam::FlowCaseType;
+    flow_case_enum flow_case_type = this->all_param.flow_solver_param.flow_case_type;
+
     dealii::GridGenerator::hyper_cube(*grid, domain_left, domain_right, colorize);
+    if(flow_case_type == flow_case_enum::sod_shock_tube
+         || flow_case_type == flow_case_enum::leblanc_shock_tube
+         || flow_case_type == flow_case_enum::shu_osher_problem)
+    {
+        for (auto cell = grid->begin_active(); cell != grid->end(); ++cell) {
+            // Set a dummy boundary ID
+            cell->set_material_id(9002);
+            if (cell == grid->begin_active())
+            {
+                for (unsigned int face=0; face<dealii::GeometryInfo<dim>::faces_per_cell; ++face) {
+                    if (flow_case_type == flow_case_enum::shu_osher_problem) {
+                        if (cell->face(face)->at_boundary()) cell->face(face)->set_boundary_id(1004);
+                    }
+                    else {
+                        if (cell->face(face)->at_boundary()) cell->face(face)->set_boundary_id(1001);
+                    }
+                }
+            }
+            else if (cell == grid->end())
+            {
+                for (unsigned int face=0; face<dealii::GeometryInfo<dim>::faces_per_cell; ++face) {
+                    if (cell->face(face)->at_boundary()) cell->face(face)->set_boundary_id (1001);
+                }
+            }
+        }
+    }
     grid->refine_global(number_of_refinements);
```

Comment:
Created a separate file for non_periodic_cube_flow grid generation.

Reply:
Could you rename the new file `non_periodic_cube_flow` --> `non_periodic_cube`? This would be more consistent with other grid names, and pedantically the grid doesn't have any flow! 

Thanks for moving this to grids.

Reply:
Renamed the files to `non_periodic_cube`! 😄